overlord/snapstate/snapmgr.go

```diff
@@ -634,6 +637,21 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+func (m *SnapManager) doSwitchSnap(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapsup, snapst, err := snapSetupAndState(t)
+	if err != nil {
+		return err
+	}
+	snapst.Channel = snapsup.Channel
```

Comment:
My naive expectation was that this would actually switch my system to that channel *and revision* if I had it on my system. I understand this is not the case but I still find this slightly confusing.

Reply:
For that we have both the refresh and the revert operations already. The whole point of the switch command is being able to alter the metadata without this causing an immediate action on the system.